wrap/src/hpx_wrap.cpp

```diff
@@ -107,7 +107,6 @@ extern "C" int initialize_main(int argc, char** argv)
         hpx::init_params iparams;
         iparams.desc_cmdline = desc;
         iparams.cfg = cfg;
-        iparams.mode = hpx::runtime_mode_console;
```

Comment:
What is the rationale of this change?

Reply:
To have the wrapping use the default runtime mode. `runtime_mode_console` tries to start the distributed runtime, but if it's disabled that will fail. I'd rather have it fall back to the local runtime just like `init` or `start` do. If the distributed runtime is enabled and the runtime mode is `runtime_mode_default` the command line handling will anyway update it to `runtime_mode_console`. There's probably a corner case I'm not aware of though. @NK-Nikunj would you happen to know why the runtime mode was explicitly set to `runtime_mode_console` here?

Reply:
To have the wrapping use the default runtime mode. `runtime_mode_console` tries to start the distributed runtime, but if it's disabled that will fail. I'd rather have it fall back to the local runtime just like `init` or `start` do. If the distributed runtime is enabled and the runtime mode is `runtime_mode_default` the command line handling will anyway update it to `runtime_mode_console`. There's probably a corner case I'm not aware of though. @NK-Nikunj would you happen to know why the runtime mode was explicitly set to `runtime_mode_console` here?

Reply:
@msimberg I explicitly set it to `runtime_mode_console` because @hkaiser asked me to take the parameters from the `init_globally` example. There was not much thought put into the config. But it may get tricky to have user defined runtime modes in the config rather than a single explicit mode.

Reply:
Thanks @NK-Nikunj! I don't think we need user defined configs in `hpx_wrap.cpp` in the sense that user code would set the runtime mode. `runtime_mode_default` will do the same without any changes from the user, but will let users pass `--hpx:connect` etc. on the command line to change the mode if desired.